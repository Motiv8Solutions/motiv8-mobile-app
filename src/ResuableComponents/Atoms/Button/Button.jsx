import React from 'react';
import styled from 'styled-components';

/**
 * Renders a button component.
 */
export class Button extends React.Component {
    constructor () {
        super();
        this.clickHandler = this.clickHandler.bind(this);
    }

    render () {
        const {label, onClick, ...rest} = this.props;
        return (
            <button onClick={this.clickHandler} {...rest}>{label}</button>
        );
    }

    clickHandler () {
        if (typeof this.props.onClick === 'function') {
            this.props.onClick(this.props.content);
        }
    }
}

export default styled(Button)`
`;