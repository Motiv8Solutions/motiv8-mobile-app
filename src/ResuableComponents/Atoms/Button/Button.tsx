import * as React from 'react';
import styled from 'styled-components';

/**
 * Renders a button component.
 */
export class Button extends React.Component<any, any> {
    constructor (props: any) {
        super(props);
        this.clickHandler = this.clickHandler.bind(this);
    }

    render () {
        const {label, onClick, ...rest} = this.props;
        return (
            <button onClick={this.clickHandler} {...rest}>{label}</button>
        );
    }

    clickHandler () {
        if (typeof this.props.onClick === 'function') {
            this.props.onClick(this.props.content);
        }
    }
}

export default styled(Button)`
`;