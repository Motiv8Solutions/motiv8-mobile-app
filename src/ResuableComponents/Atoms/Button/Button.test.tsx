import Button from './Button';
import * as renderer from 'react-test-renderer';
import * as React from 'react';

test('Button test', () => {
    const component = renderer.create(
        <Button/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});