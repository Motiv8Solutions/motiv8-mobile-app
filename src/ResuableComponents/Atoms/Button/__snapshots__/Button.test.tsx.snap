// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test 1`] = `
<button
  className="sc-bdVaJa fePDys"
  onClick={[Function]}
/>
`;
